File: javascript/acronym/acronym.js
//
// This is only a SKELETON file for the 'Acronym' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const parse = phrase => 
    phrase.toUpperCase().split(/[\ \-]/g).map(word => word.match(/[A-Z]/)).join('');

File: cpp/acronym/acronym.cpp
#include<string>
#include "acronym.h"

namespace acronym {
    std::string acronym(std::string phrase){
        std::string acronym;
        acronym += toupper(phrase[0]);

        for(int i = 0; i != (int)phrase.length(); i++){
            if(phrase[i] == ' ' || phrase[i] == '-')
                acronym += toupper(phrase[i + 1]);
        }
        return acronym;
    }
}  // namespace acronym
